vcpkg/users/buildsystems/cmake-integration.md

```diff
@@ -12,13 +12,35 @@ See [Installing and Using Packages Example: sqlite](../../examples/installing-an
 
 Projects configured to use the vcpkg toolchain file (via the CMake setting `CMAKE_TOOLCHAIN_FILE`) can find libraries from vcpkg using the standard CMake functions: `find_package()`, `find_path()`, and `find_library()`.
 
+We recommend using [CMake Presets] to specify your toolchain file. For example, if you have defined the environment variable `VCPKG_ROOT`, you can use the following `CMakePresets.json` and pass `--preset debug` on the configure line:
+
+:::code language="json" source="snippets/cmake-integration/CMakePresets.json":::
+
+```sh
+cmake -B build -S /my/project --preset debug
+```
+
+If you need to use an absolute path for vcpkg specific to your current machine, you can use `CMakeUserPresets.json` and add it to your `.gitignore` file.
+
+:::code language="json" source="snippets/cmake-integration/CMakeUserPresets.json":::
+
+CMake versions older than 3.19 must pass the toolchain file on the configure command line:
```

Comment:
Or use `cacheVariables` map

Reply:
CMakePresets support was introduced in 3.19, so older than that must pass it on the command line :). Technically, 3.19 and 3.20 can pass it via `cacheVariables` but not `toolchainFile`. I didn't feel it was helpful to exhaustively enumerate all the possibilities (the `CMAKE_TOOLCHAIN_FILE` environment variable, for example) -- so I stuck to "Recent users should do X, older users should do Y".